pkg/auth/requests/sar/sar_test.go

```diff
@@ -0,0 +1,135 @@
+package sar
```

Comment:
sure! I added tests just for `UserCanImpersonateExtras` because is the only new method. `UserCanImpersonateUser` and `UserCanImpersonateGroup` were already there. But you are right, it'd be nice to also add tests for those. I'll do that

Reply:
tests added in https://github.com/rancher/rancher/pull/46968/commits/0b8a5464c1ab1c527df39c35811c60fd7a7d3392